parsers/UY.py

```diff
@@ -1,163 +1,224 @@
-
-    current_time = arrow.now("UTC-3")
-    if current_time.minute < 30:
-        # Data for current hour seems to be available after 30mins.
-        current_time = current_time.shift(hours=-1)
-    lookup_time = current_time.floor("hour").format("DD/MM/YYYY HH:mm")
-
-    s = session or Session()
-    url = SALTO_GRANDE_URL + lookup_time
-    response = s.get(url)
-    soup = BeautifulSoup(response.text, "html.parser")
-
-    tie = soup.find("div", style="position:absolute; top:143; left:597")
-    generation = float(tie.text)
-
-    return generation
-
 
-def parse_page(session: Optional[Session]):
-    r = session or Session()
-    url = "https://apps.ute.com.uy/SgePublico/ConsPotenciaGeneracionArbolXFuente.aspx"
-    response = r.get(url)
-    soup = BeautifulSoup(response.text, "html.parser")
 
-    datefield = soup.find(
-        "span", attrs={"id": "ctl00_ContentPlaceHolder1_lblUltFecScada"}
+def get_adme_url(target_datetime: datetime, session: Optional[Session] = None) -> str:
+    """"""
+    session = session or Session()
+    next_day = target_datetime + timedelta(days=1)
+    date_format = "{day}%2F{month}%2F{year}".format(
+        day=target_datetime.day, month=target_datetime.month, year=target_datetime.year
     )
-    datestr = re.findall("\d\d/\d\d/\d\d\d\d \d+:\d\d", str(datefield.contents[0]))[0]
-    date = arrow.get(datestr, "DD/MM/YYYY h:mm").replace(tzinfo=dateutil.tz.gettz(tz))
-
-    table = soup.find(
-        "table", attrs={"id": "ctl00_ContentPlaceHolder1_gridPotenciasNivel1"}
+    next_day_format = "{day}%2F{month}%2F{year}".format(
+        day=next_day.day, month=next_day.month, year=next_day.year
     )
 
-    obj = {"datetime": date.datetime}
-
-    for tr in table.find_all("tr"):
-        tds = tr.find_all("td")
-        if not len(tds):
-            continue
-
-        key = tds[0].find_all("b")
-        # Go back one level up if the b tag is not there
-        if not len(key):
-            key = tds[0].find_all("font")
-        k = key[0].contents[0]
-
-        value = tds[1].find_all("b")
-        # Go back one level up if the b tag is not there
-        if not len(value):
-            value = tds[1].find_all("font")
-        v_str = value[0].contents[0]
-        if v_str.find(",") > -1 and v_str.find(".") > -1:
-            # there can be values like "1.012,5"
-            v_str = v_str.replace(".", "")
-            v_str = v_str.replace(",", ".")
-        else:
-            # just replace decimal separator, like "125,2"
-            v_str = v_str.replace(",", ".")
-        v = float(v_str)
-
-        # solar reports -0.1 at night, make it at least 0
-        v = max(v, 0)
-
-        obj[k] = v
-
-    # https://github.com/tmrowco/electricitymap/issues/1325#issuecomment-380453296
-    salto_grande = get_salto_grande(session)
-    obj["Hidráulica"] = obj.get("Hidráulica", 0.0) + salto_grande
-
-    return obj
-
-
+    link = ADME_URL + date_format + "&fecha_fin=" + next_day_format + "&send=MOSTRAR"
```

Comment:
```suggestion
    link = f"ADME_URL{date_format}&fecha_fin={next_day_format}&send=MOSTRAR"
```
It's probably a personal opinion, I tend to find f-strings more readable than concatenation.